### PR TITLE
LPS-137216 Adjustments for Collection Pagination

### PR DESCRIPTION
**Pull request by veroglez**

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/Collection.js [hunk 1]
```diff
@@ -168,23 +168,25 @@ const ColumnContext = ({
...
-	items: [{}],
-	length: 1,
-	totalNumberOfItems: 1,
-};
-
 const Collection = React.forwardRef(
 	({children, item, withinTopper = false}, ref) => {
 		const child = React.Children.toArray(children)[0];
 		const collectionConfig = item.config;
+		const emptyCollection = useMemo(() => ({
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/Collection.js]: I know it's designed that way in the Figma, but for me it's a little bit weird to show the alert five times if we have five items configured for an empty collection. Maybe it would make more sense to show only one alert, regardless of the configured number of items. What do you think? Maybe we can discuss it with Mery 😄

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/CollectionPagination.js [hunk 2]
```diff
@@ -73,7 +73,7 @@ export default function CollectionPagination({
 						activePage={activePage}
 						onPageChange={onPageChange}
 						totalPages={
-							Number.isFinite(totalPages) ? totalPages : 0
+							(Number.isFinite(totalPages) && totalPages) || 1
```
Comment (sandrodw3) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/CollectionPagination.js:75]: Just curiosity, why did you change the operands? Just for esthetic? 😛
IMHO the ternary if looks more clear.

**Review by sandrodw3 — COMMENTED** [review]
Nice work! 🥰
The refactor to set error messages in the useEffects makes the logic more clear 👏🏽
I just left some tiny comments, but they are more like suggestions. Apart of that, in my opinion it's good to go!

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/CollectionPagination.js (hunk 2 above)
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/CollectionPagination.js:75]: It's because sometimes the `totalPage` has the value of 0, so with the ternary the condition `Number.isFinite (totalPages)` would still be `true`.

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/Collection.js (hunk 1 above)
Comment (veroglez) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/layout-data-items/Collection.js]: Yes, I agree, I think we can discuss it with her 👍

**Comment by sandrodw3** [comment]
Resent with extra commits here: https://github.com/liferay-echo/liferay-portal/pull/5608
